streams/src/main/java/org/apache/kafka/streams/Topology.java

```diff
@@ -677,10 +662,10 @@ public synchronized <KIn, VIn, KOut, VOut> Topology addProcessor(final String na
      *
      * @throws TopologyException
      *         if the {@link StoreBuilder#name() state store} was already added, or
-     *         if a processor name is unknown
+     *         if a processor name is unknown or specifies a source or sink
```

Comment:
I did find the check for "sinks as parent nodes" when processors and sink are added. Realized that adding stores to sources/sinks does not make sense either -- added new checks in internal classed for it.

Reply:
Good call